src/libraries/System.Private.CoreLib/src/System/SpanHelpers.Byte.cs

```diff
@@ -385,24 +386,26 @@ public static bool Contains(ref byte searchSpace, byte value, int length)
 
             while (lengthToExamine > 0)
             {
-                lengthToExamine -= 1;
+                lengthToExamine--;
 
                 if (uValue == Unsafe.AddByteOffset(ref searchSpace, offset))
                     goto Found;
 
-                offset += 1;
+                offset++;
             }
 
-            if (Vector.IsHardwareAccelerated && (offset < (nuint)(uint)length))
+            if (Vector.IsHardwareAccelerated && (offset < (uint)length))
             {
-                lengthToExamine = (((nuint)(uint)length - offset) & (nuint)~(Vector<byte>.Count - 1));
+                lengthToExamine = ((uint)length - offset) & (nuint)~(Vector<byte>.Count - 1);
 
-                Vector<byte> values = new Vector<byte>(value);
+                Vector<byte> zero = Vector<byte>.Zero;  // JIT won't hoist that vector outside the loop
+                Vector<byte> values = new(value);
+                Vector<byte> matches;
 
-                while (lengthToExamine > offset)
+                while (offset < lengthToExamine)
                 {
-                    var matches = Vector.Equals(values, LoadVector(ref searchSpace, offset));
-                    if (Vector<byte>.Zero.Equals(matches))
+                    matches = Vector.Equals(values, LoadVector(ref searchSpace, offset));
+                    if (zero.Equals(matches))
```

Comment:
looks like there is CQ issue in this pattern:
![image](https://user-images.githubusercontent.com/523221/161429379-64b64977-c623-4532-a665-9d161fd52a34.png)

(feel free to file an issue)

Reply:
More over you don't need to hoist it - it should not be used

Reply:
Ah, `vec1 == vec2` emits better code.

For the char-overload:
```diff
M01_L05:
       vpcmpeqw  ymm2,ymm1,[rcx+r9*2]
-      vpxor     ymm2,ymm0,ymm2
       vptest    ymm2,ymm2
       jne       short M01_L08
       add       r9,10
       cmp       r9,r11
       jb        short M01_L05
M01_L06:
       cmp       r9,r10
       jae       short M01_L07
       vpcmpeqw  ymm2,ymm1,[rcx+r10*2+0FFE0]
-      vpxor     ymm0,ymm0,ymm2
       vptest    ymm0,ymm0
       jne       short M01_L08
```

Thanks for the hint!
Will create an issue for that.